artview/components/mapper.py

```diff
@@ -0,0 +1,298 @@
+            ("refl_filter_flag", bool),
+            ("refl_field", str),
+            ("max_refl", float),
+            ("map_roi", bool),
+            ("weighting_function", ("Barnes", "Cressman")),
+            ("toa", float),
+            ]
+        self.roi_parameters_type = [
+            ("roi_func", ("constant", "dist", "dist_beam")),
+            ("constant_roi", float),
+            ("z_factor", float),
+            ("xy_factor", float),
+            ("min_radius", float),
+            ("bsp", float)
+            ]
+        self.gridding_parameters_type = [
+            ("copy_field_data", bool),
+            ("algorithm", ("kd_tree", "ball_tree")),
+            ("leafsize", int),
+            ]
+
+        if Vradar is None:
+            self.Vradar = Variable(None)
+        else:
+            self.Vradar = Vradar
+
+        if Vgrid is None:
+            self.Vgrid = Variable(None)
+        else:
+            self.Vgrid = Vgrid
+        self.sharedVariables = {"Vradar": self.NewRadar,
+                                "Vgrid": None}
+        self.connectAllVariables()
+
+        self.NewRadar(None, True)
+        self.show()
+
+    def mountUI(self):
+        '''Mount Options Layout.'''
+
+        self.despeckleButton = QtWidgets.QPushButton("Map")
+        self.despeckleButton.clicked.connect(self.grid_from_radars)
+        self.layout.addWidget(self.despeckleButton, 7, 0, 1, 2)
+
+        parentdir = os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                                 os.pardir))
+        config_icon = QtGui.QIcon(os.sep.join(
+            [parentdir, 'icons', "categories-applications-system-icon.png"]))
+        self.configButton = QtWidgets.QPushButton(config_icon,"")
+        self.layout.addWidget(self.configButton, 7, 2)
+        self.configMenu = QtWidgets.QMenu(self)
+        self.configButton.setMenu(self.configMenu)
+
+        self.configMenu.addAction(QtWidgets.QAction("Set General Parameters", self,
+                                                triggered=self.setParameters))
+        self.configMenu.addAction(QtWidgets.QAction("Set Radius of Influence Parameters", self,
+                                                triggered=self.setRoiParameters))
+        self.configMenu.addAction(QtWidgets.QAction("Set map_to_grid Parameters", self,
+                                                triggered=self.setGriddingParameters))
+        self.fieldMenu = self.configMenu.addMenu("Fields")
+        self.configMenu.addAction(QtWidgets.QAction("Help", self,
+                                                triggered=self._displayHelp))
+
+        self.layout.addWidget(QtWidgets.QLabel("Z"), 1, 0, 2, 1)
+        self.layout.addWidget(QtWidgets.QLabel("Y"), 3, 0, 2, 1)
+        self.layout.addWidget(QtWidgets.QLabel("X"), 5, 0, 2, 1)
+
+        self.gridShapeZ = QtWidgets.QSpinBox()
+        self.gridShapeZ.setRange(0, 1000000)
```

Comment:
I would at minimum define these range values as a variable that can be used instead of copying the same numbers. Also, should this be a keyword option for the user?